### PR TITLE
[SPARK-11988] [ML] [MLLIB] Update JPMML to 1.2.7

### PR DESCRIPTION
**Pull request by srowen**
Update JPMML pmml-model to 1.2.7

### CODE REVIEW

File: mllib/pom.xml [hunk 1]
```diff
@@ -109,7 +109,7 @@
     <dependency>
       <groupId>org.jpmml</groupId>
       <artifactId>pmml-model</artifactId>
-      <version>1.1.15</version>
+      <version>1.2.7</version>
```
Comment (srowen) [inline mllib/pom.xml]: @vruusmann I've just realized this module is published as `pmml-model` and `jpmml-model`. Which is standard or preferred?

**Comment by vruusmann** [comment]
The `jpmml-model` is the top-level (ie. project) artifact. It provides "cover" for a list of modules, such as `pmml-model`, `pmml-schema`, `pmml-agent` etc.

If you want to include PMML class model then a single `pmml-model` dependency will suffice. If you declared `jpmml-model` dependency, then you would get a bunch of unnecessary modules included in your build.

File: mllib/pom.xml (hunk 1 above)
Comment (vruusmann) [inline mllib/pom.xml]: The `jpmml-model` is the top-level (ie. project) artifact. It provides "cover" for a list of modules, such as `pmml-model`, `pmml-schema`, `pmml-agent` etc.

If you want to include PMML class model then a single `pmml-model` dependency will suffice. If you declared `jpmml-model` dependency, then you would get a bunch of unnecessary modules included in your build.

**Comment by srowen** [comment]
I'm inclined to merge this for 1.6 for reasons in the JIRA, but am leaving this open a bit longer.

**Comment by srowen** [comment]
To be conservative, I'd like to merge this but only to master. Would that be OK? see further discussion about dependencies in JIRA

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**